### PR TITLE
docs: create first draft of dataspace protocol architecture documents

### PR DESCRIPTION
**Pull request by jimmarino**
This is a draft PR for documents outlining the architecture of EDC IDS protocol support.

## Linked Issue(s)

Part of #2429

### CODE REVIEW

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
Still not finished the review, but I think as we already want to go forward my comments could eventually already be valuable.

File: docs/developer/architecture/ids-next/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# IDS-Next Architecture
```
[the remark this answers is not in the document]
Comment (jimmarino) [reply docs/developer/architecture/ids-next/README.md:1]: DRs will come later and follow these documents. The DRs will lay out specific decisions that have an architectural impact; these documents lay out the overall architecture.

File: docs/developer/architecture/ids-next/contract-negotiation-architecture.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+Note this is does not follow immutability principles but has the advantage of simplicity and the ability to enforce proper state transitions in the `ContractNegotiation` instance.
+The `Result<Boolean>` response indicates if the `ContractNegotiation` has been updated (true) or not (false), or if processing failed. This can be used to avoid
+re-persisting `ContractNegotiation` instances if they have not changed.
+
+End users will be able to register their own `StateTransitionFunction` implementations to provide custom contract negotiation workflows. For example, a `StateTransitionFunction`
+implementation can be registered to be invoked when a consumer request is received (CONSUMER_REQUESTED) to support manual approval. The function will be called by
+the `ConsumerContractNegotiationManager` during its processing loop. If the request initiates a new negotiation, the function implementation could trigger a process in an external
+approval system. In a subsequent processing loop, the `StateTransitionFunction` could check for the completion of the external approval process (e.g. by querying a column in a
+table) and updating the `ContractNegotition`.
```
[the remark this answers is not in the document]
Comment (jimmarino) [reply docs/developer/architecture/ids-next/contract-negotiation-architecture.md:91]: I specifically wanted to call out the use of a table for correlation here.

File: docs/developer/architecture/ids-next/ids-endpoints-services-architecture.md [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+```
+
+Similarly, a `ToCatalogTransformer` can convert from a JSON=LD structure to the EDC `Catalog` type:
+
+```
+public class ToCatalogTransformer extends AbstractJsonLdTransformer<JsonObject, Catalog> {
+    private static final String DCAT_CATALOG = "http://www.w3.org/ns/dcat/Catalog";
+    private static final String DCAT_DATASET = "http://www.w3.org/ns/dcat/dataset";
+    private static final String DCAT_DISTRIBUTION = "http://www.w3.org/ns/dcat/distribution";
+    private static final String DCAT_DATA_SERVICE = "http://www.w3.org/ns/dcat/DataService";
```
[the remark this answers is not in the document]
Comment (jimmarino) [reply docs/developer/architecture/ids-next/ids-endpoints-services-architecture.md:168]: Some could be

File: docs/developer/architecture/ids-dataspace-protocol/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# IDS Dataspace Protocol Architecture
+
+The EDC will be upgraded to implement and be fully compliant to the __IDS Next__ protocol specifications. These include:
```
Comment (juliapampus) [inline docs/developer/architecture/ids-dataspace-protocol/README.md:3]: "IDS Next" is still mentioned in all of your documents.

File: docs/developer/architecture/ids-dataspace-protocol/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# IDS Dataspace Protocol Architecture
```
Comment (juliapampus) [inline docs/developer/architecture/ids-dataspace-protocol/README.md:1]: Without "IDS"

**Review by juliapampus — CHANGES_REQUESTED** [review]

**Review by ndr-brt — APPROVED** [review]
All good, I'm a little confused because IDS and everyone always wrote about "data space" and now we're calling it "dataspace" without the white space. For that matter, I prefer the latter.

**Review by juliapampus — APPROVED** [review]

**Pull request merged**